packages/client/src/exchanges/types/exchange.types.ts

```diff
@@ -0,0 +1,53 @@
+import type { Order, OrderItem } from '../../orders/index.js';
+import type { PaymentIntent } from '../../payments/index.js';
+import type { ProductResult, ProductVariant } from '../../products/index.js';
+import type { Return } from '../../returns/index.js';
+
+export type Exchange = {
+  id: string;
+  status: ExchangeStatus;
+  exchangeOrder: ExchangeOrder;
+  customer: { id: number | null };
+  exchangeGroups: Array<ExchangeGroup>;
+  createdDate: string;
+  updatedDate: string | null;
+};
+
+export enum ExchangeStatus {
+  Draft = 'Draft',
+  PendingOrderConfirmation = 'PendingOrderConfirmation',
+  Booked = 'Booked',
+  ProcessingOrderPayment = 'ProcessingOrderPayment',
+  PendingOrderDelivery = 'PendingOrderDelivery',
+  Cancelled = 'Cancelled',
+  Completed = 'Completed',
+}
+
+export type ExchangeOrder = {
+  // This specific checkoutOrderId is a string on purpose (usually it is a number).
+  checkoutOrderId: string;
+  orderId: Order['id'];
+  payment: { intentId: PaymentIntent['id'] };
+};
+
+export type ExchangeGroup = {
+  exchangeReturnItems: Array<ExchangeReturnItem>;
+  exchangeItems: Array<ExchangeItem>;
+};
+
+export type ExchangeReturnItem = {
+  id?: string;
```

Comment:
Why is `id` an optional here? This should be required as it is used in response values and it should be optional only on request values.